### ISSUE TITLE
Keep inner curly brace block opener on same line as case that opened it

### ISSUE THREAD

**Issue by fishb6nes** [scalameta/scalafmt#1518]
I'm looking to keep inner curly brace blocks on the same line as the match case they apply to. A little bit like https://github.com/scalameta/scalafmt/issues/1126, but for case statements.

https://github.com/scalameta/scalafmt/issues/1454 has the same problem, but sadly has gone unanswered. Is there a configuration or custom rewrite rule that would allow the following?:

- **Configuration**:
```
version = "2.0.1"
newlines.alwaysBeforeMultilineDef = false
newlines.alwaysBeforeTopLevelStatements = true
```

## Steps

Given code like this:
```scala
/* ... inside a larger match block */
case it: SomeClass => it.someEnumValue match {
  case EnumValue.some => "someResult"
}
```

When I run scalafmt on file save in IntelliJ

## Problem

Scalafmt formats code like this:
```scala
/* ... inside a larger match block */
case it: SomeClass =>
  it.someEnumValue match {
    case EnumValue.some => "someResult"
  }
```

## Expectation

I would like the formatted output to look like this: (unchanged from original)
```scala
/* ... inside a larger match block */
case it: SomeClass => it.someEnumValue match {
  case EnumValue.some => "someResult"
}
```

Thank you for taking a look.

**Comment by tanishiking** [comment]
@fishb6nes Thank you for reporting!
Unfortunately, I'm not sure is there an option for realizing the result...
It would be reasonable to keep inner curly brace blocks on the same line, as long as the line length doesn't exceed the `maxColumn` 👍

**Comment by kitbellew** [comment]
@fishb6nes @fjunk @stefan-hoehn @kiendang @harpocrates @wdanilo in #1793, the following formatting is implemented for the new `newlines.source=fold` option:
```
/* ... inside a larger match block */
case it: SomeClass => it.someEnumValue match {
    case EnumValue.some => "someResult"
  }
case etc...
```
compared to your original, it's *indented*, to be consistent with indenting of case body relative to case clause, and to differentiate from the next case keyword. let me know if you'd like this to be added to the classic formatting (i.e., not specifying `newlines.source`).

**Comment by fishb6nes** [comment]
Sweet! Thank you for the heads up. Personally I am more than happy with the opt-in flag.

**Comment by kitbellew** [comment]
this change has been released in 2.5.0, as part of the new `newlines.source=fold` managed formatting mode. if that's sufficient, please close the issue.

**Comment by fishb6nes** [comment]
Thank you Albert. I can confirm that it works with `newlines.source = fold`, however that opt-in flag changes a _lot_ of formatting rules. Would it be possible to also have it work with `newlines.source = keep`? Where if the user has folded the match statements they are left folded (while within maxColumn and other rules), and if the user has unfolded the match statements they are left unfolded.

Currently `newlines.source = keep` seems to always unfold nested match statements.
